ee/tabby-webserver/src/hub/api.rs

```diff
@@ -196,4 +198,41 @@ impl RepositoryAccess for SchedulerClient {
     async fn list_repositories(&self) -> Result<Vec<RepositoryConfig>> {
         Ok(self.0.list_repositories(Context::current()).await?)
     }
+
+    fn process_file(&self, version: u64, file: SourceFile) {
+        let cache = match RepositoryCache::new() {
+            Ok(cache) => cache,
+            Err(e) => {
+                error!("Failed to open repository cache: {e}");
+                return;
+            }
+        };
+        let file_path = file.filepath.clone();
+        if let Err(e) = cache.add_repository_meta(version, file.into()) {
+            error!("Failed to write {} to repository cache: {e}", file_path);
+        };
+    }
+
+    fn finish_snapshot(&self, version: u64) {
+        let cache = match RepositoryCache::new() {
+            Ok(cache) => cache,
+            Err(e) => {
+                error!("Failed to open repository cache: {e}");
+                return;
+            }
+        };
+        let old_version = match cache.latest_version() {
+            Ok(v) => v,
+            Err(e) => {
+                error!("Failed to get next repository cache version: {e}");
+                return;
+            }
+        };
+        if let Err(e) = cache
+            .set_version(version)
+            .and_then(|_| cache.clear_versions_under(old_version))
```

Comment:
```suggestion
        if let Err(e) = cache.update_latest_version(version)
```

in update_latest_version, set new version and remove old versions